ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -81,13 +82,15 @@
   /* Is discovery enabled? */
   private boolean isActive = false;
   protected final Subscribers<PeerBondedObserver> peerBondedObservers = Subscribers.create();
+  private final MaintainedPeers maintainedPeers;
```

Comment:
A simpler way of implementing this would be to call the publicly exposed method [`PeerDiscoveryAgent.bond()`](https://github.com/hyperledger/besu/blob/0455a8d937541136adc0b079e169db9ea451e9a8/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java#L310-L312) from [`P2PNetwork.addMaintainConnection`](https://github.com/hyperledger/besu/blob/0455a8d937541136adc0b079e169db9ea451e9a8/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java#L265-L270).